retrofit/src/main/java/retrofit/MethodHandler.java

```diff
@@ -28,7 +29,9 @@
     Converter<ResponseBody, Object> responseConverter =
         (Converter<ResponseBody, Object>) createResponseConverter(method, retrofit, responseType);
     RequestFactory requestFactory = RequestFactoryParser.parse(method, responseType, retrofit);
-    return new MethodHandler<>(retrofit, requestFactory, callAdapter, responseConverter);
+    String infoForException = collectInfoForException(method);
```

Comment:
Will try to make it lazy. But it may require sharing some data: `method` or `MethodHandler` with the `Call`, that's why I didn't go that road in the initial impl.